src/components/admin/SpacesDropdown/SpacesDropdown.tsx

```diff
@@ -69,75 +77,45 @@ export const SpacesDropdown: React.FC<SpacesDropdownProps> = ({
     [filteredSpaces]
   );
 
-  const spaceOptions = useMemo(() => [spaceNoneOption, ...sortedSpaces], [
-    sortedSpaces,
-  ]);
+  const remapSpaces = sortedSpaces.map((space) => {
+    const spaceIcon = space?.host?.icon
+      ? space?.host?.icon
+      : PORTAL_INFO_ICON_MAPPING[space.template ?? ""];
+    return {
+      id: space.id,
+      label: space.name,
+      template: space.template,
+      icon: spaceIcon,
+    };
+  });
+
+  const spaceOptions: SpacesOption[] = useMemo(
+    () => [spaceNoneOption, ...remapSpaces],
+    [remapSpaces]
+  );
 
   useEffect(() => {
     if (parentSpace) {
       setSelected(parentSpace);
     }
   }, [parentSpace]);
 
-  const renderedOptions = useMemo(
-    () =>
-      spaceOptions.map(({ id, name, template, host }) => {
-        const spaceIcon = PORTAL_INFO_ICON_MAPPING[template ?? ""];
-
-        return (
-          <div
-            key={id}
-            onClick={() => {
-              setSelected({ name, template, id });
-              setValue(fieldName, id, { shouldValidate: true });
-            }}
-            data-dropdown-value={name}
-            className="flex items-center w-max"
-          >
-            {name !== spaceNoneOption.name ? (
-              <img
-                alt={`space-icon-${spaceIcon}`}
-                src={host?.icon || spaceIcon}
-                className="w-6 h-6 mr-2 rounded-full"
-              />
-            ) : null}
-            {name || noneOptionName}
-          </div>
-        );
-      }) ?? ALWAYS_EMPTY_ARRAY,
-    [spaceOptions, setValue, fieldName]
-  );
-
-  const renderedTitle = useMemo(() => {
-    if (!selected) {
-      return <span data-dropdown-value="">Select a space</span>;
-    }
-
-    const space = spaces?.[selected.id ?? ""] ?? parentSpace;
-
-    const spaceIcon = PORTAL_INFO_ICON_MAPPING[space?.template ?? ""];
-
-    return (
-      <span data-dropdown-value={selected.name}>
-        {selected.name !== spaceNoneOption.name ? (
-          <img
-            alt={`space-icon-${spaceIcon}`}
-            src={spaceIcon}
-            className="w-6 h-6 mr-2 rounded-full"
-          />
-        ) : null}
-        {selected.name || noneOptionName}
-      </span>
-    );
-  }, [spaces, selected, parentSpace]);
+  const select = ({ label, id, icon, template }: SpacesOption) => {
+    setSelected({ name: label, id, icon, template });
+    setValue(fieldName, id, { shouldValidate: true });
+  };
 
   return (
     <>
       <div data-bem="SpacesDropdown" className="mb-2">
         <label className="block text-sm font-medium text-gray-700 mb-1">
           Link portal to a space
         </label>
-        <Dropdown title={renderedTitle}>{renderedOptions}</Dropdown>
+        <Dropdown
+          title={selected?.name || noneOptionName}
+          options={spaceOptions}
+          onSelect={(space) => select(space)}
```

Comment:
will that work?
```suggestion
          onSelect={select}
```

Reply:
Yes, it will pass the parameters automatically